controllers/scopeinstance_controller.go

```diff
@@ -338,6 +358,47 @@ func (r *ScopeInstanceReconciler) deleteBindings(ctx context.Context, listOption
 	return nil
 }
 
+// deleteOldBindings will delete any (Cluster)RoleBindings that are owned by the given ScopeInstance and are no longer up to date.
+func (r *ScopeInstanceReconciler) deleteOldBindings(ctx context.Context, in *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate) error {
+	siHashReq, err := labels.NewRequirement(scopeInstanceHashKey, selection.NotEquals, []string{util.HashObject(in.Spec)})
+	if err != nil {
+		return err
+	}
+
+	siUIDReq, err := labels.NewRequirement(scopeInstanceUIDKey, selection.Equals, []string{string(in.GetUID())})
+	if err != nil {
+		return err
+	}
+
+	listOptions := &client.ListOptions{
+		LabelSelector: labels.NewSelector().Add(*siHashReq, *siUIDReq),
+	}
+
+	if err := r.deleteBindings(ctx, listOptions); err != nil {
+		return err
+	}
+
+	stHashReq, err := labels.NewRequirement(scopeTemplateHashKey, selection.NotEquals, []string{util.HashObject(st.Spec)})
+	if err != nil {
+		return err
+	}
+
+	stUIDReq, err := labels.NewRequirement(scopeTemplateUIDKey, selection.Equals, []string{string(st.GetUID())})
+	if err != nil {
+		return err
+	}
+
+	listOptions = &client.ListOptions{
+		LabelSelector: labels.NewSelector().Add(*siUIDReq, *stUIDReq, *stHashReq),
+	}
+
+	if err := r.deleteBindings(ctx, listOptions); err != nil {
+		return err
+	}
+
+	return nil
+}
+
```

Comment:
I think we could clean this up into 2 separate functions if we feel it would be cleaner. 

The reason why there have to be 2 separate `deleteBindings` calls is because we want to delete all bindings where the label that stores the `ScopeTemplate` hash is different OR where the label that stores the `ScopeInstance` hash is different.

Unfortunately for label requirements like this Kubernetes only supports AND for label selectors: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors

Reply:
> I think we could clean this up into 2 separate functions if we feel it would be cleaner.
> 
> The reason why there have to be 2 separate `deleteBindings` calls is because we want to delete all bindings where the label that stores the `ScopeTemplate` hash is different OR where the label that stores the `ScopeInstance` hash is different.
> 
> Unfortunately for label requirements like this Kubernetes only supports AND for label selectors: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors

Yeah, I keep thinking of SQL and databases where you can do an OR :)

Reply:
In commit https://github.com/operator-framework/oria-operator/pull/48/commits/e99a979253cb50b7abf599e7d6764a84fdcf9f01 I updated the comment above the function and added some in the function to explain exactly what is happening. 

I also added a TODO to potentially explore converting the separate hashes for the `ScopeInstance.Spec` and `ScopeTemplate.Spec` fields into a single hash. This would make it so we could have that OR type logic as the hash would be different if either of them changed. Before I went that route though I wanted to see what thoughts on this were, so WDYT?